jiant/tasks/tasks.py

```diff
@@ -3731,3 +3732,160 @@ def get_metrics(self, reset=False):
+            "train": os.path.join(path, "train.txt"),
+            "val": os.path.join(path, "valid.txt"),
+            "test": os.path.join(path, "test.txt"),
+        }
+        self._label_namespace = self.name + "_labels"
+
+    def get_target_seq_length(self):
+        target_is_max = random.random() > 0.1
+        if target_is_max:
+            target_seq_length = self.max_seq_len
+        else:
+            target_seq_length = random.randint(2, self.max_seq_len)
+        return target_seq_length
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text. We override the
+        this function and all functions that call this function because
+        the step of reading in the data for SOP is different than other
+        PairClassificationTasks.
+
+        ALBERT does SOP classification by, for each document:
+            For each example:
+                -90% of the time, getting sentences of max_seq_length number of tokens, and
+                10% of the time, getting sentences of a random number of tokens between 2 and max_seq_length.
+                -Given the sampled sentences, randomly sample N such that the first N sentences in the 
+                sampled go to the first segment, and the rest go to the second. 
+                -50% of the time, the first and second segments are switched. 
+            There is also some logic here to make sure that we always split by the sentence instead of 
+            in the middle of a sentence.
+        Args:
+            path: (str) data file path
+        """
+        f = open(path, "r")
+        #  The dataset comes with one sentence per line, thus we split by
+        #  line here.
+        current_chunk = [tokenize_and_truncate(self._tokenizer_name, next(f), -1)]
+        current_length = 0
+        target_seq_length = self.get_target_seq_length()
+        in_order = 1
+        while len(current_chunk) > 0:
+            segment = next(f)
+            segment = tokenize_and_truncate(self._tokenizer_name, segment, -1)
+            if "END OF ARTICLE" in segment or current_length >= target_seq_length:
+                for_next_chunk = []
+                if "END OF ARTICLE" not in segment:
+                    for_next_chunk.append(segment)
+                if current_length > target_seq_length:
+                    # Since the most current sentence added to the chunk exceeds the target
+                    # length, we save it for the next chunk (next example).
+                    for_next_chunk.append(current_chunk[-1])
+                    current_chunk.pop(-1)
+                target_seq_length = self.get_target_seq_length()
+                if len(current_chunk) >= 2:
+                    # Make sure we have at least 2 sentences to distribute between the two
+                    # segments.
+                    a_end = 1
+                    if len(current_chunk) > 2:
+                        a_end = random.randint(1, len(current_chunk) - 1)
```

Comment:
```python
a_end = random.randint(1, len(current_chunk) - 1)
```` should suffice, right?

We know from the above that `len(current_chunk) >= 2`. If `len(current_chunk)==2`, we get `random.randint(1, 1)` which is always 1.